include/unifex/let_error.hpp

```diff
@@ -33,89 +34,89 @@
+template <typename Source, typename Func, typename Receiver>
 struct _op {
   class type;
 };
-template <typename Source, typename Error, typename Receiver>
-using operation_type = typename _op<Source, Error, remove_cvref_t<Receiver>>::type;
+template <typename Source, typename Func, typename Receiver>
+using operation_type = typename _op<Source, Func, remove_cvref_t<Receiver>>::type;
 
-template <typename Source, typename Error, typename Receiver>
+template <typename Source, typename Func, typename Receiver>
 struct _rcvr {
   class type;
 };
-template <typename Source, typename Error, typename Receiver>
-using receiver_type = typename _rcvr<Source, Error, remove_cvref_t<Receiver>>::type;
+template <typename Source, typename Func, typename Receiver>
+using receiver_type = typename _rcvr<Source, Func, remove_cvref_t<Receiver>>::type;
 
-template <typename Source, typename Error, typename Receiver>
+template <typename Source, typename Func, typename Receiver, typename Error>
 struct _frcvr {
   class type;
 };
-template <typename Source, typename Error, typename Receiver>
-using final_receiver_type = typename _frcvr<Source, Error, remove_cvref_t<Receiver>>::type;
+template <typename Source, typename Func, typename Receiver, typename Error>
+using final_receiver_type =
+    typename _frcvr<Source, Func, remove_cvref_t<Receiver>, Error>::type;
 
-template <typename Source, typename Error>
+template <typename Source, typename Func>
 struct _sndr {
   class type;
 };
 
-template <typename Source, typename Error>
-using _sender = typename _sndr<Source, Error>::type;
+template <typename Source, typename Func>
+using _sender = typename _sndr<Source, Func>::type;
 
-enum class state { neither, source, final };
-
-template <typename Source, typename Error, typename Receiver>
-class _rcvr<Source, Error, Receiver>::type final {
-  using operation = operation_type<Source, Error, Receiver>;
-  using final_receiver = final_receiver_type<Source, Error, Receiver>;
-  using final_sender_t = callable_result_t<Error&>;
+template <typename Source, typename Func, typename Receiver>
+class _rcvr<Source, Func, Receiver>::type final {
+  using operation = operation_type<Source, Func, Receiver>;
+  template <typename Error>
+  using final_receiver = final_receiver_type<Source, Func, Receiver, Error>;
 
 public:
-  explicit type(operation* op) noexcept
-  : op_(op) {}
+  explicit type(operation* op) noexcept : op_(op) {}
+  type(type&& other) noexcept : op_(std::exchange(other.op_, {})) {}
 
-  type(type&& other) noexcept
-  : op_(std::exchange(other.op_, {}))
-  {}
- 
   template(typename... Values)
     (requires receiver_of<Receiver, Values...>)
   void set_value(Values&&... values) noexcept(
       is_nothrow_receiver_of_v<Receiver, Values...>) {
     UNIFEX_ASSERT(op_ != nullptr);
+    unifex::deactivate_union_member(op_->sourceOp_);
     unifex::set_value(std::move(op_->receiver_), (Values&&)values...);
   }
```

Comment:
These operations cannot be easily flipped - `op_` may be destroyed as a side-effect of calling `set_value` (e.g. because the final receiver has already destroyed the full state of the operation). I changed the types in `set_value()` to `Values... values`, similar to how `set_error()` accepts error by value in `let_value` combinator